### PR TITLE
Redesign status line modification

### PR DESCRIPTION
**Pull request by SlimaneAmar**
PR must be merged after PR : #64

### CODE REVIEW

File: src/main/resources/db/changelog/changesets/changelog_2021-12-24T14:47:39Z.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-3.10.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.10.xsd">
+    <changeSet author="amarsli (generated)" id="1640357270577-5">
+        <createTable tableName="branch_status_modification">
+            <column name="action" type="VARCHAR(255)"/>
+            <column name="id" type="UUID">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="branch_status_modificationPK"/>
+            </column>
+        </createTable>
+    </changeSet>
```
Comment (FranckLecuyer) [inline src/main/resources/db/changelog/changesets/changelog_2021-12-24T14:47:39Z.xml]: Don't know why liquibase generates these subsequent changesets on two windings transformer creation foreign keys ...
Maybe, you can remove them ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 2]
```diff
@@ -82,14 +82,18 @@ public NetworkModificationController(NetworkModificationService networkModificat
 
     @PutMapping(value = "/networks/{networkUuid}/lines/{lineId}/status", consumes = MediaType.TEXT_PLAIN_VALUE)
     @Operation(summary = "Change the status of a line in a network variant")
-    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "the line status has been changed")})
-    public ResponseEntity<Flux<EquipmenModificationInfos>> changeLineStatus(
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "the line status has been changed"),
+        @ApiResponse(responseCode = "404", description = "the network or line not found"),
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:93]: ?

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java [hunk 3]
```diff
@@ -80,4 +86,9 @@ public static NetworkModificationException createEquipmentTypeUnknown(String typ
         Objects.requireNonNull(type);
         return new NetworkModificationException(Type.UNKNOWN_EQUIPMENT_TYPE, "The equipment type : " + type + " is unknown");
     }
+
+    public static NetworkModificationException createBranchActionTypeUnknown(String type) {
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:87]: Done

File: src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java [hunk 4]
```diff
@@ -297,126 +291,125 @@ public void testNetworkOrVariantNotFound() {
...
+    }
 
-        webTestClient.put().uri(uriString, TEST_NETWORK_ID, "line3")
-                .bodyValue("trip")
-                .exchange()
-                .expectStatus().is5xxServerError()
-                .expectBody(String.class)
-                .isEqualTo(new NetworkModificationException(MODIFICATION_ERROR, "Unable to disconnect both line ends").getMessage());
+    @Test
+    public void testLineStatusMofificationWithErrors() {
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java:343]: Done

File: src/main/resources/db/changelog/changesets/changelog_2021-12-24T14:47:39Z.xml (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/resources/db/changelog/changesets/changelog_2021-12-24T14:47:39Z.xml]: ?

File: src/main/java/org/gridsuite/modification/server/entities/BranchStatusModificationEntity.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ */
+@NoArgsConstructor
+@Getter
+@Entity
+@Table(name = "branchStatusModification")
+@PrimaryKeyJoinColumn(foreignKey = @ForeignKey(name = "branchStatusModification_id_fk_constraint"))
+public class BranchStatusModificationEntity extends ModificationEntity {
+
+    @Column(name = "action")
+    private String action;
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/entities/BranchStatusModificationEntity.java]: Good idea !
I did not know

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java [hunk 6]
```diff
@@ -79,19 +81,19 @@ Type getType() {
         return type;
     }
 
-    public static NetworkModificationException createEquipmentTypeUnknown(String type) {
+    public static NetworkModificationException createEquipmentTypeUnknown(@NonNull String type) {
         Objects.requireNonNull(type);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:85]: with the @NonNull annotation, you can remove Objects.requireNonNull call

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java [hunk 7]
```diff
@@ -79,19 +81,19 @@ Type getType() {
...
-        return new NetworkModificationException(Type.BRANCH_ACTION_TYPE_UNKNOWN, "The branch action type : " + type + " is unknown");
+        return new NetworkModificationException(Type.BRANCH_ACTION_TYPE_UNSUPPORTED, "The branch action type : " + type + " is unsupported");
     }
 
-    public static NetworkModificationException createBranchActionTypeBad(String type) {
+    public static NetworkModificationException createBranchActionTypeUnknown(@NonNull String type) {
         Objects.requireNonNull(type);
-        return new NetworkModificationException(Type.BRANCH_ACTION_TYPE_BAD, "The branch action type : " + type + " is bad");
+        return new NetworkModificationException(Type.BRANCH_ACTION_TYPE_UNKNOWN, "The branch action type : " + type + " is bad");
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:96]: 'unknown' instead of 'bad'

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/changesets/changelog_2021-12-24T14:47:39Z.xml (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/resources/db/changelog/changesets/changelog_2021-12-24T14:47:39Z.xml]: These additiional changesets were present in the first version of the file ...
But they disapeared in the second version, so it's ok ...

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:93]: Done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java (hunk 7 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:96]: Done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationException.java (hunk 6 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationException.java:85]: Done

**Review by FranckLecuyer — APPROVED** [review]

**Review by niconoir — APPROVED** [review]

**Pull request merged**